apps/sync/js/adapters/bookmarks.js

```diff
@@ -305,7 +322,10 @@ DataAdapters.bookmarks = {
         type: payload.type === 'bookmark' ? 'url' : 'others',
         iconable: false,
         icon: '',
-        fxsyncPayload: payload,
+        fxsyncRecord: {
+          last_modified: remoteRecords[i].last_modified,
+          payload: remoteRecords[i].payload
+        },
         fxsyncId: payload.id
       });
     }
```

Comment:
It will be more like this for pushing a record:
```javascript
      var fxsyncRecords = {};
      fxsyncRecords[payload.id] = remoteRecords[i].payload;
      fxsyncRecords[payload.id].last_modified = remoteRecords[i].last_modified;
      bookmarks.push({
        // URL is the ID for bookmark records in bookmarks_store, but there are
        // some types without a valid URL except bookmark type. URL is used as
        // its ID to compatible bookmarks_store for bookmark type record.
        // The combination of type and fxsyncID is used as its ID for the types
        // except bookmark.
        id: payload.type === 'bookmark' ? payload.bmkUri :
          (payload.type + '|' + payload.id),
        url: payload.bmkUri,
        name: payload.title,
        type: payload.type === 'bookmark' ? 'url' : 'others',
        iconable: false,
        icon: '',
        fxsyncRecords: fxsyncRecords
      });
```